packages/server/src/routes/api/login.ts

```diff
@@ -0,0 +1,94 @@
+import config from '../../config';
+import Router from '../../utils/Router';
+import { redirect, SubPath } from '../../utils/routeUtils';
+import { generateRedirectHtml, identityProvider, serviceProvider } from '../../utils/saml';
+import { AppContext, RouteType, SamlPostResponse } from '../../utils/types';
+import { bodyFields } from '../../utils/requestUtils';
+import { InternalServerError } from '../../utils/errors';
+import { FlowResult } from 'samlify/types/src/flow';
+import { cookieSet } from '../../utils/cookies';
+import defaultView from '../../utils/defaultView';
+
+export const router = new Router(RouteType.Api);
+
+router.public = true;
+
+// Set an error message saying that SAML is disabled.
+// @param ctx The AppContext.
+function samlNotAvailable(ctx: AppContext) {
+	ctx.status = 403;
+	ctx.body = { error: 'This server does not accept SAML authentication.' };
+}
+
+// Redirect the user to the IdP login page, if they somehow get to this URL directly.
+router.get('api/saml', async (_path: SubPath, ctx: AppContext) => {
+	if (config().saml.enabled) {
+		return await generateRedirectHtml();
+	} else {
+		return samlNotAvailable(ctx);
+	}
+});
+
+// Called when a user successfully authenticated with the IdP, and was redirected to Joplin.
+router.post('api/saml', async (_path: SubPath, ctx: AppContext) => {
+	if (config().saml.enabled) {
```

Comment:
At the top just have `if (config().saml.enabled) throw new ErrorForbidden('SAML not enabled')`. So we lose one level of indent